### ISSUE TITLE
Submenu with Combobox Demo has bad aria

### ISSUE THREAD

**Issue by derekjackson-das** [ariakit/ariakit#4443]
### Current behavior

In the Submenu with Combobox [demo](https://urldefense.proofpoint.com/v2/url?u=https-3A__ariakit.org_examples_menu-2Dnested-2Dcombobox&d=DwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=lSjxiQ4YiczCXKRXsY6srn3D4DulY2OVXH4Xg0n6cYw&m=ZoCiTSyKKYW3H8x7RvxKKC0hjj56q4t2YXKSfNOd8f9XyU9_9vLWpi2_zJ6UUm0a&s=CKxXaWUFhqa8QQaGIEa-6Jw5ZrgA1uoLVN_Ub9Bi6-k&e=), the aria-haspopup attribute changes its value when the combo-box is opened. Currently it shows aria-haspopup=“menu” and then changes to aria-haspopup=“dialog” when opened. Also, the popup menu has the role=“listbox”. The popup’s role should match the aria-haspopup’s value. Here is the [screen recording](https://urldefense.proofpoint.com/v2/url?u=https-3A__share.zight.com_geuQvv0d&d=DwMFaQ&c=WO-RGvefibhHBZq3fL85hQ&r=lSjxiQ4YiczCXKRXsY6srn3D4DulY2OVXH4Xg0n6cYw&m=ZoCiTSyKKYW3H8x7RvxKKC0hjj56q4t2YXKSfNOd8f9XyU9_9vLWpi2_zJ6UUm0a&s=I7SqMntIwJuzoRQIBRQw0ieYzd-ef17CS-nteNePLnU&e=) showing these issues.

I don't know if this just a bad implementation on the demo or a fault of the component but either should be fixed to make sure the component is implemented in the most accessible manner. The aria-haspopup attribute shouldn’t change when the menu is open or closed, and it should match the role of the component that it is opening. So if it is opening a menu, the role of the component should be menu, and the aria-haspopup value should be menu. Here is the [W3C spec](https://w3c.github.io/aria/#aria-haspopup) with more details on how it should be implemented and the values allowed, and corresponding roles it can be applied too.

### Steps to reproduce the bug

1. Open https://ariakit.org/previews/menu-nested-combobox
2. Open inspector tools or use a screen reader to control the button and listen to the description of the button.
3. See the aria-haspopup value in the inspector is "menu" or listen to the screen reader description of the button as "menu popup, button".
4. Click the button and observe in the inspector window the value of the aria-haspopup value has changed to dialog or navigate with the screen reader back to the button and listen to the description as "dialog popup, button".
~~5. With the popup menu open inspect the menu container and you can see the role attribute does not match either values (menu or dialog) of the aria-haspopup attribute. Screen readers will describe it as a listbox.~~

### Expected behavior

The aria-haspopup attribute will not change its value and will accurately describe the role of the component it controls (allowed values are: false (default), true, menu, listbox, tree, grid, or dialog)

### Workaround

https://github.com/ariakit/ariakit/issues/4443#issuecomment-2749494527
https://github.com/ariakit/ariakit/issues/4443#issuecomment-4196389266

**Comment by derekjackson-das** [comment]
Thanks! Updated the description.

**Comment by diegohaz** [comment]
Workaround: Add `aria-haspopup="dialog"` to `MenuButton`:

```jsx
<Ariakit.MenuButton aria-haspopup="dialog" ...
```
